scripts/travis_prepare.sh

```diff
@@ -114,20 +114,22 @@ create () {
 
 # Would give a permission denied error in the travis build dir
 cd /
-
-# Postgres versions supported by Travis CI
-if (( ! "$DONT_TEST_PRESENT" )); then
```

Comment:
You don't like a double negative I see? :smile:

Reply:
They are in travis ci: more options (top right) -> settings.

You should have all of them, except the protected ones. But maybe you don't and you need them?

The way they were defined was so that they do a reasonable thing if you have them all to 0, hence the negative to test the present. Do you think it's too awkward? We can set `TEST_PRESENT=1` in the defaults.

Honestly I would make this even more robust by using the `-u` switch in the bash script so that the script fails in case of undefined variable, and set the sensible default into the script in order to be explicit in what is its default behaviour and what can be overridden.

What do you think?

![image](https://user-images.githubusercontent.com/199429/87958636-b00ac100-caa9-11ea-8850-a4a85d38e145.png)

Reply:
> They are in travis ci: more options (top right) -> settings.

 I think only you can set them. I am not able to see them.

> You should have all of them, except the protected ones. But maybe you don't and you need them?

Having all of them sounds good.
 
> The way they were defined was so that they do a reasonable thing if you have them all to 0, hence the negative to test the present. Do you think it's too awkward? We can set `TEST_PRESENT=1` in the defaults.

Please use `TEST_PRESENT=1` as it looks better than testing with a double negative.

> Honestly I would make this even more robust by using the `-u` switch in the bash script so that the script fails in case of undefined variable, and set the sensible default into the script in order to be explicit in what is its default behaviour and what can be overridden.

Yes, we can make setting of these variables mandatory.

Reply:
In this case probably the script was working by relying that whatever wasn't set was equivalent to 0. It's not the way I wrote bash scripts nowadays, I got a bit stricter.

What I would like to see is the script running with `set -u` (throwing errors if a variable is undefined) and having those 5 variables defined inside the script with a pattern of "set them to '0' if they are undefined", which I think you can obtain with something like 

    export TEST_PAST=${TEST_PAST:-0}

and `TEST_PRESENT`, defaulting to 1, replacing the double negative one.

To organise: is it something you can try to do or do you want me to do it? It may take me some time to be able to jump on this task though.

Reply:
I have added it to the global variables section in `travis.yml`. It is not working as expected if we add them in `travis_prepare.sh` and `travis_test.sh`. Tests will pass once you set `TEST_PAST=0` in Travis-CI settings or remove the variables from Travis-CI settings. Please check the local build [report](https://travis-ci.org/github/odidev/psycopg2/builds/710266079) for confirmation on test execution.